test/hooks/useSelector.spec.js

```diff
@@ -412,6 +412,35 @@ describe('React', () => {
 
           spy.mockRestore()
         })
+
+        it('reuse latest selected state on selector re-run', () => {
+          store = createStore(({ count } = { count: -1 }) => ({
+            count: count + 1,
+          }))
+
+          const alwaysEqual = () => true
```

Comment:
Does this also maintain the latest state if the equalityFn is defined within the component or inline?

Reply:
Yes—it only cares what the equalityFn returns, not what its identity is

Reply:
However, having a non-pure equalityFn (i.e., one that closes over variables from its surrounding components) could have *very surprising results.